src/encoded/static/components/browse/components/AboveTableControls/SelectedFilesDownloadButton.js

```diff
@@ -0,0 +1,248 @@
+                <span className="hidden-xs hidden-sm">Download </span>
+                <span className="count-to-download-integer">{ countToShow }</span>
+                <span className="hidden-xs hidden-sm text-400"> Selected Files</span>
+            </SelectedFilesDownloadButton>
+        );
+    }
+}
+
+
+
+/**
+ * Upon clicking the button, reveal a modal popup giving users more download instructions.
+ */
+export class SelectedFilesDownloadButton extends React.PureComponent {
+
+    static totalSelectedFilesCount = memoize(function(selectedFiles){ return _.keys(selectedFiles || {}).length; });
+
+    static propTypes = {
+        'windowWidth' : PropTypes.number.isRequired,
+        'id' : PropTypes.string,
+        'selectedFiles' : PropTypes.object.isRequired,
+        'filenamePrefix' : PropTypes.string.isRequired
+    };
+
+    static defaultProps = {
+        'id' : null,
+        'filenamePrefix' : "metadata_",
+        'children' : "Download",
+        'className' : 'btn-primary'
+    };
+
+    constructor(props){
+        super(props);
+        this.hideModal = this.hideModal.bind(this);
+        this.showModal = this.showModal.bind(this);
+        this.state = { 'modalOpen' : false };
+    }
+
+    hideModal(){
+        this.setState({ 'modalOpen' : false });
+    }
+
+    showModal(){
+        this.setState({ 'modalOpen' : true });
+    }
+
+    render(){
+        var { selectedFiles, filenamePrefix, children, disabled, selectedFilesUniqueCount } = this.props;
+        const btnProps = _.omit(this.props, 'filenamePrefix', 'selectedFiles', 'windowWidth', 'children', 'selectedFilesUniqueCount', 'disabled');
+        const isDisabled = disabled || SelectedFilesDownloadButton.totalSelectedFilesCount(selectedFiles) === 0;
+        return (
+            <React.Fragment>
+                <Button {...btnProps} disabled={isDisabled} onClick={this.showModal}>
+                    { children }
+                </Button>
+                { this.state.modalOpen ?
+                    <SelectedFilesDownloadModal {...{ selectedFiles, filenamePrefix, selectedFilesUniqueCount }} onHide={this.hideModal}/>
+                : null }
+            </React.Fragment>
+        );
+    }
+}
+
+
+
+class SelectedFilesDownloadModal extends React.PureComponent {
+
+    static findUnpublishedFiles = memoize(function(selectedFiles){
+        return _.any(_.values(selectedFiles || {}), function(file){
+            return !(
+                (file.from_experiment && file.from_experiment.from_experiment_set && file.from_experiment.from_experiment_set.produced_in_pub)
+                || (file.from_experiment_set && file.from_experiment_set.produced_in_pub)
+            );
+        });
+    });
+
+    constructor(props){
+        super(props);
+        this.handleAcceptDisclaimer = this.handleAcceptDisclaimer.bind(this);
+        this.state = {
+            'disclaimerAccepted' : false
```

Comment:
Cleaned up state. `disclaimerAccepted` is the only state variable that should be needed -- other things/values can be derived directly from it.

Adding more state variables than needed is discouraged as state management is more complex than functional patterns/code.